suit_generator/cmd_payload_extract.py

```diff
@@ -0,0 +1,70 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+"""CMD_PAYLOAD_EXTRACT CLI command entry point."""
+
+import cbor2
+import logging
+
+log = logging.getLogger(__name__)
+
+PAYLOAD_EXTRACT_CMD = "payload_extract"
+
+def add_arguments(parser):
+    """Add additional arguments to the passed parser."""
+    cmd_payload_extract_arg_parser = parser.add_parser(PAYLOAD_EXTRACT_CMD, help="Create raw cache structure.")
+
+    cmd_payload_extract_arg_parser.add_argument(
+        "--input-envelope",
+        required=True,
+        help="Input envelope file path."
+    )
+    cmd_payload_extract_arg_parser.add_argument(
+        "--output-envelope",
+        required=True,
+        help="Output envelope file path."
+    )
+    cmd_payload_extract_arg_parser.add_argument(
+        "--payload-name",
+        required=True,
+        help="Name of the integrated payload to extract."
+    )
+    cmd_payload_extract_arg_parser.add_argument("--output-payload-file", required=False, help="Output payload file path to store the extracted payload."
+                                                + "If not provided, the payload will not be stored to a file.")
+
+    cmd_payload_extract_arg_parser.add_argument(
+        "--payload-replace-name",
+        help="Name of the integrated payload to replace the extracted payload with."
+            + "If not provided, the payload will be removed from the envelope."
+    )
+
+def main(input_envelope: str, output_envelope: str, payload_name: str, output_payload_file: str, payload_replace_name: str) -> None:
+    """Extract an integrated payload from a SUIT envelope.
+
+    :param input_envelope: input envelope file path
+    :param output_envelope: output envelope file path
+    :param payload_name: name of the integrated payload to extract
+    :param output_payload_file: output file path to store the extracted payload
+        None if the payload should not be stored to a file
+    :param payload_replace_name: name of the integrated payload to replace the extracted payload with.
+        None if the payload should be removed from the envelope.
+    """
+    envelope = None
```

Comment:
The `envelope` value is immediately assigned in line 56 as `envelope = cbor2.load(fh)`, so the initial assignment is redundant and could be removed

Reply:
Fixed